openedx/features/course_experience/views/course_dates.py

```diff
@@ -26,3 +27,16 @@ def render_to_fragment(self, request, course_id=None, **kwargs):
         }
         html = render_to_string('course_experience/course-dates-fragment.html', context)
         return Fragment(html)
+
+
+class CourseDatesFragmentMobileView(CourseDatesFragmentView):
```

Comment:
Is this new subclass necessary? It would be simpler to just have a single fragment view that can be used by everybody. Why should it 404 for unauthenticated users? We have been moving towards making course views visible to everyone for SEO reasons. @marcotuts, what do you think?

Reply:
I would hope that we could use the same view and minimize drift here if possible.

Reply:
Actually the original view render less dates for unauthenticated users which actually does not meet mobile app requrements, the scenario here is that if user is authenticated with mobile app and visit this fragment then user will see less dates because he/she is not looged in on our server `courses.edx.org` and at that time mobile app does not know user is logged in or not on our server. 
That's why we are returning 404 to unauthenticated users which is already handled in mobile app to authenticate users on our server, does that make sense?

CC: @verdan, @marcotuts

Reply:
@andy-armstrong on the mobile clients we are using WebKit to create and maintain a session with the server for authenticated requests. But unfortunately, WebKit hasn't exposed any way to find out either session was created with the server or not, so we have implemented a mechanism to automatically create/recreate session with the server for all authenticated requests if the server returns 404 and loads underlying request after creating the session. 

This mechanism was initially implemented to load xBlocks on mobile, but now we are using the same piece of the fragment to load all authenticated requests. It would be helpful if the server could return 404 instead of showing limited data (when the user is logged out, the existing fragment displays limited dates/data), otherwise, we won't be able to differentiate if the system is displaying limited dates or the actual i.e., complete dates. 

cc: @marcotuts

Reply:
Thanks for the explanation @saeedbashir. I don't love that we're going to have to duplicate every view that we have that doesn't throw a 404, but it makes sense for now. We can't make the fragment throw a 404 for the non-mobile case because that fragment is shown to users who access the course without being logged in.

Reply:
@waheedahmed Can you add this explanation as part of the docstring so that folks can understand why this was necessary.

Reply:
Sure.